### ISSUE TITLE
Rotate 3D axis labels

### ISSUE THREAD

**Issue by alexcjohnson** [plotly/plotly.js#3077]
Long axis titles on 3D scenes overlap the tick labels and data, because they're simply centered on a point outside the axis and not rotated:
![screen shot 2018-10-04 at 3 54 01 pm](https://user-images.githubusercontent.com/2678795/46499366-dfc1d980-c7ed-11e8-8b32-70cdf4879a24.png)
Our test images don't include a lot of long 3d titles, but you can see this effect a little bit in [gl3d_world-cals](https://github.com/plotly/plotly.js/blob/master/test/image/baselines/gl3d_world-cals.png)

Feels like the easiest solution - which is also consistent with our 2D axis titles - would be to rotate these labels parallel to the axis. But potentially 1-letter titles ("X" etc) should remain unrotated?

The degree of overlap also depends on the viewpoint, so the logic for where to put this relative to the axis line and tick labels probably needs a bit of adjustment:
![3d title overlap](https://user-images.githubusercontent.com/2678795/46499814-1f3cf580-c7ef-11e8-8c55-6b570435fbe3.gif)

**Comment by etpinard** [comment]
We could probably merge https://github.com/plotly/plotly.js/issues/511 with this new ticket.

**Comment by alexcjohnson** [comment]
good call. I'll close #511. Referencing http://community.plot.ly/t/axes-labels-overlapping-on-3d-surface-plot/859

**Comment by alexcjohnson** [comment]
a `titleangle` attribute would let people (manually) fix the overlap for one particular camera angle, but as soon as the user starts rotating the scene, the overlap returns. The two options I see are (1) the title rotates to align with the axis, or (2) the title stays horizontal but changes to right-aligned on the left side and left-aligned on the right side. (1) is my strong preference as it's more compact (ie less likely to overflow the scene), and also helps guide your eye as to the representation of that variable.

**Comment by MeganBeckett** [comment]
Hi there! Is this now possible? ie. to make long axis titles lie parallel to the axes in 3d plots? It seems something has been developed, but I can't see how to implement in plotly (I'm also using plot_ly in R)?

**Comment by archmoj** [comment]
@MeganBeckett 
Yes. The latest versions of plotly R/js/py do support auto rotation by default. 
Also in case you may be interested in line breaks or using text formatting in 3D scenes, here is an example: https://rreusser.github.io/plotly-mock-viewer/#gl3d_formatted-text-on-multiple-lines
